applications/visualizer/frontend/src/helpers/twoD/coloringStrategy/CellTypeColoringStrategy.ts

```diff
@@ -0,0 +1,52 @@
+import {ColoringStrategy, ColorMap} from "./ColoringStrategy.ts";
+import {Neuron} from "../../../rest";
+
+export enum CellType {
+    Neurosecretory = 'neurosecretory',
+    Sensory = 'sensory',
+    Inter = 'inter',
+    Motor = 'motor',
+    Muscle = 'muscle',
+    Others = 'others'
+}
+
+const cellTypeColors: ColorMap = {
```

Comment:
I think your code here is overly complicated. Why not create an object to represent a celltype, which would include colors...etc, and instanciate a simple js object with all the combinations? I'm not sure I see the advantage of having the singleton and the lazy init, what you are doing with spliting between the celltypecolor, the celltypemap is splitting the inner components of an OO object, while the strategy here is simulating the behavior, more or less, of an object, trying to compose again values that are supposed to go together. 

If we push a little bit further, we could have simply a js object that would have this head:

```js
enum CellType {
	...
}

interface CellDesign = {
	type: CellType,
	color: string
}

const cellsConfig = {
  i: {type: CellType.Inter, color: '#FF887A'},
  b: {type: CellType.Muscle, color: ....},
... // etc    
}
```

Another option, as we don't have an at run time extensible set of cell types, you can have a superclass and subclasses for each types. 

For the strategy, I feel the singleton is unecessary here and it makes even less sense also considering that it already accesses `const` object that is declared as global. Usually using singleton is always controversial, and the point of the singleton is to avoid a global state, which is not really the case here in a way. The fact of using an enum to represent the two strategy and switching from one to the other using a function and relating the enum value to one or the other singleton looks like a misinterpretation of the strategy pattern (or weird implementation), but I understand the constraint as the menu needs to select one of the options.